kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/InternalUtils.java

```diff
@@ -169,4 +167,10 @@ public static Path relativizePath(Path child, URI root) {
   public static Set<String> toLowerCaseSet(Collection<String> set) {
     return set.stream().map(String::toLowerCase).collect(Collectors.toSet());
   }
+
+  public static <T> List<T> toList(CloseableIterator<T> iterator) {
+    List<T> result = new ArrayList<>();
+    iterator.forEachRemaining(result::add);
+    return result;
+  }
```

Comment:
Ugh sorry hope this is the last thing, but @scottsand-db added `CloseableIterator::toInMemoryList` can we use that and remove this method?